src/Common/ArrayCache.h

```diff
@@ -179,13 +179,22 @@ class ArrayCache : private boost::noncopyable
         {
             ptr = mmap(address_hint, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
             if (MAP_FAILED == ptr)
-                throw ErrnoException(DB::ErrorCodes::CANNOT_ALLOCATE_MEMORY, "Allocator: Cannot mmap {}", ReadableSize(size));
+                throw DB::ErrnoException(DB::ErrorCodes::CANNOT_ALLOCATE_MEMORY, "Allocator: Cannot mmap {}", ReadableSize(size));
         }
 
         ~Chunk()
         {
             if (ptr && 0 != munmap(ptr, size))
-                throw ErrnoException(DB::ErrorCodes::CANNOT_MUNMAP, "Allocator: Cannot munmap {}", ReadableSize(size));
+            {
+                try
+                {
+                    throw DB::ErrnoException(DB::ErrorCodes::CANNOT_MUNMAP, "Allocator: Cannot munmap {}", ReadableSize(size));
+                }
+                catch (DB::ErrnoException &)
+                {
+                    DB::tryLogCurrentException(__PRETTY_FUNCTION__);
+                }
```

Comment:
Why are you throwing a function just to catch it?

Reply:
For logging. If we didn't catch it the most likely result will be program termination (as the caller won't catch it and directly).

Reply:
Ah you want to log the stacktrace?
Okay, I assume there isn't any other way to do it.

Reply:
Well, throwing the exception also adds an error to system.errors (and it's a real / important error), so throwing is a nice way to handle everything.